connectors/wsclient/src/main/java/quarks/connectors/wsclient/runtime/WebSocketClientReceiver.java

```diff
@@ -0,0 +1,45 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2015,2016 
+*/
+package quarks.connectors.wsclient.runtime;
+
+import quarks.function.Consumer;
+import quarks.function.Function;
+
+public class WebSocketClientReceiver<T> implements Consumer<Consumer<T>>, AutoCloseable {
+    private static final long serialVersionUID = 1L;
+    protected final WebSocketClientConnector connector;
+    private final Function<String,T> toTuple;
+    protected Consumer<T> eventHandler;
+    
+    public WebSocketClientReceiver(WebSocketClientConnector connector, Function<String,T> toTuple) {
+        this.connector = connector;
+        this.toTuple = toTuple;
+    }
+
+    @Override
+    public void accept(Consumer<T> eventHandler) {
+        this.eventHandler = eventHandler;
+        connector.setReceiver(this);
```

Comment:
This accept is the setup callback from Topology.events() and it's only called once for the lifetime of the job to initialize the "source".  If setReceiver() was called prior to receiving the eventHandler, there might be cases where the receiver would be given a message so it would have to check every time to ensure it had an eventHandler to give it to.